src/parser/index.ts

```diff
@@ -0,0 +1,59 @@
+import { IllegalArgumentException, IllegalStateException } from "./errors";
+import { Node } from "./node";
+import { Resource, Resources } from "./resources";
+import { Scene } from "./scene";
+import { Section } from "./section";
+
+export class AbstractSyntaxTree extends Node {
```

Comment:
This should probably be in its own file.
Typically `index.ts` is for indexing the folder,
```
export { Node } from "./node";
export { Scene } from "./scene";
...
```